Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeModel.cpp

```diff
@@ -262,13 +262,20 @@ namespace AssetProcessor
                 return true;
             });
 
-        
+
         AZStd::vector<ProductDependencyChild> pendingChildren;
 
         m_sharedDbConnection->QueryDirectReverseProductDependenciesBySourceGuidSubId(
             assetId.m_guid, assetId.m_subId,
             [&](AzToolsFramework::AssetDatabase::ProductDatabaseEntry& incomingDependency)
             {
+                // Make sure we haven't already encountered this product before
+                // If we have, that means there's a dependency loop, so just abort
+                if (visitedDependencies.contains(incomingDependency.m_productID))
+                {
+                    return true;
```

Comment:
Would it help the user to issue a warning if a dependency loop is encountered?

Reply:
There's actually other, unrelated code which is doing this already